src/main/java/net/fabricmc/loader/impl/discovery/RuntimeModRemapper.java

```diff
@@ -213,5 +232,38 @@ private static class RemapInfo {
 		OutputConsumerPath outputConsumerPath;
 		String accessWidenerPath;
 		byte[] accessWidener;
+
+		private boolean requiresRemap() throws IOException {
+			final Path manifestPath = inputPath.resolve("META-INF/MANIFEST.MF");
+
+			if (Files.notExists(manifestPath)) return true;
+
+			try (InputStream is = Files.newInputStream(manifestPath)) {
+				final Manifest manifest = new Manifest(is);
+				final String namespace = manifest.getMainAttributes().getValue("Fabric-Mapping-Namespace");
+				final String loomRemap = manifest.getMainAttributes().getValue("Fabric-Loom-Remap");
```

Comment:
Im a little unsure if we want to technically add this to loader API 🤔 might be best to just leave it.